lib/interceptor/dns.js

```diff
@@ -171,24 +165,24 @@ class DNSInstance {
       return ip
     }
 
-    const timestamp = Date.now()
-    // Record TTL is already in ms
-    if (ip.timestamp != null && timestamp - ip.timestamp > ip.ttl) {
+    if (Date.now() - ip.timestamp > ip.ttl) { // record TTL is already in ms
       // We delete expired records
       // It is possible that they have different TTL, so we manage them individually
       family.ips.splice(position, 1)
       return this.pick(origin, hostnameRecords, affinity)
     }
 
-    ip.timestamp = timestamp
-
     this.lastIpFamily = newIpFamily
     return ip
   }
 
-  setRecords (origin, addresses) {
+  setRecords (origin, opts, addresses) {
+    const timestamp = Date.now()
     const records = { records: { 4: null, 6: null } }
     for (const record of addresses) {
+      record.timestamp = timestamp
+      record.ttl = opts.maxTTL
```

Comment:
If the record already contains a `ttl` let's preserve it, otherwise default to `maxTTL`.

On top, let's use the min from both values.

Reply:
I have added it.

How do you feel about expecting the record result to be in ms? `dns.lookup` doesn't usually provide ttl. `dns.resolve` does, but it's ttl, and other time properties, are in seconds.
https://nodejs.org/api/dns.html#dnsresolveanyhostname-callback